src/search.c

```diff
@@ -69,19 +69,23 @@ main(int argc, char *argv[])
 	i = search_makelist(res, argv[1]);
 	if (i == -1) {
 		fprintf(stderr, _("File not found\n"));
-		return 1;
+		goto out;
 	}
 	num_mirrors = search_getspeeds(res, i);
 	if (num_mirrors < 0) {
 		fprintf(stderr, _("Speed testing failed\n"));
-		return 1;
+		goto out;
 	}
 	printf(_("%i usable mirrors:\n"), num_mirrors);
 	search_sortlist(res, i);
 	for (j = 0; j < i; j++)
 		printf("%-70.70s %5i\n", res[j].url, res[j].speed);
 
-	return 0;
+out:
+	ret = 0;
```

Comment:
This makes it return always 0.

Reply:
Isn't there a way to disable it for this function?

Reply:
yeah, ret = 0 should have gone above the label.
What do you want to disable exactly?

Reply:
The leak warning for functions named `main`.

Reply:
Why? it's still an unreleased resource (harmless of course because then the process will be killed and all the resources released), but still a bad pattern that may accidentally copy/pasted around.

I'm more in favour of fixing also "apparently harmless" behaviours.